test/main.cpp

```diff
@@ -19,6 +20,15 @@ int main() {
 	qc.import(filename);
 	qc.dump("test_dump.qasm");
 
+	qc.import("./circuits/measurement_test.qasm");
+	
+	qc::CircuitOptimizer::removeFinalMeasurements(qc);
+
+	qc.dump("measurement_result.qasm");
+	qc.import("./circuits/swap_test.qasm");
+	qc::CircuitOptimizer::decomposeSWAP(qc, true);
+	qc.dump("swap_result.qasm");
+
```

Comment:
Instead of the "tests" here, it would be better to add new unit tests under `test/unittests/test_qfr_functionality.cpp`.
There are already tests for the other optimizations.
This way, all applications of the new optimizations can be tested on small examples and, additionally, code coverage is increased.
As a consequence, adding the additional circuit files should not be necessary.